common/script/ops/hourglassPurchase.js

```diff
@@ -3,53 +3,59 @@ import i18n from '../i18n';
 import _ from 'lodash';
 import splitWhitespace from '../libs/splitWhitespace';
 import pickDeep from '../libs/pickDeep';
+import {
+  BadRequest,
+  NotAuthorized,
+} from '../libs/errors';
+
+module.exports = function purchaseHourglass (user, req = {}, analytics) {
+  let key = _.get(req, 'body.key');
+  if (!key) throw new BadRequest(i18n.t('missingKeyParam', req.language));
+
+  let type = _.get(req, 'body.type');
+  if (!type) throw new BadRequest(i18n.t('missingTypeParam', req.language));
 
-module.exports = function(user, req, cb, analytics) {
-  var analyticsData, key, ref, type;
-  ref = req.params, type = ref.type, key = ref.key;
   if (!content.timeTravelStable[type]) {
-    return typeof cb === "function" ? cb({
-      code: 403,
-      message: i18n.t('typeNotAllowedHourglass', req.language) + JSON.stringify(_.keys(content.timeTravelStable))
-    }) : void 0;
+    throw new NotAuthorized(i18n.t('typeNotAllowedHourglass', req.language));
+    //  JSON.stringify(_.keys(content.timeTravelStable))
   }
+
   if (!_.contains(_.keys(content.timeTravelStable[type]), key)) {
-    return typeof cb === "function" ? cb({
-      code: 403,
-      message: i18n.t(type + 'NotAllowedHourglass', req.language)
-    }) : void 0;
+    throw new NotAuthorized(i18n.t('NotAllowedHourglass', req.language));
   }
+
   if (user.items[type][key]) {
-    return typeof cb === "function" ? cb({
-      code: 403,
-      message: i18n.t(type + 'AlreadyOwned', req.language)
-    }) : void 0;
+    throw new NotAuthorized(i18n.t('AlreadyOwned', req.language));
   }
+
   if (!(user.purchased.plan.consecutive.trinkets > 0)) {
```

Comment:
this should be user.purchased.plan.consecutive.trinkets <= 0, it's more readable IMHO